src/it/scala/djinni/GeneratorIntegrationTest.scala

```diff
@@ -10,71 +10,104 @@ class GeneratorIntegrationTest extends IntegrationTest with GivenWhenThen {
     val djinniTypes = Table(
       ("idlFile",
         "cppFilenames",
+        "cppHeaderFilenames",
         "javaFilenames",
         "jniFilenames",
+        "jniHeaderFilenames",
         "objcFilenames",
+        "objcHeaderFilenames",
         "objcppFilenames"),
       ("my_enum",
-        Cpp("my_enum.hpp"),
+        Cpp(),
+        CppHeaders("my_enum.hpp"),
         Java("MyEnum.java"),
-        Jni("my_enum.hpp"),
-        ObjC("ITMyEnum.h"),
+        Jni(),
+        JniHeaders("my_enum.hpp"),
+        ObjC(),
+        ObjCHeaders("ITMyEnum.h", "bridging-header.h"),
```

Comment:
Because I added all the `header` parameters in the test djinni call, I need to check the header & source file expectations separately. Additionally I check for the bridging-header in the `objc-headers` folder.